lib/pure/os.nim

```diff
@@ -865,6 +865,19 @@ type
 
 {.deprecated: [TPathComponent: PathComponent].}
 
+
+when defined(posix):
+  proc getSymlinkFileKind(path: string): PathComponent =
+    # Helper function.
+    var s : Stat
```

Comment:
Also, could you raise an error when ``path`` is ``""`` in this proc with a meaningful error?